src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -265,6 +265,7 @@ uct_cuda_ipc_md_open(uct_component_t *component, const char *md_name,
         .mkey_pack          = uct_cuda_ipc_mkey_pack,
         .mem_reg            = uct_cuda_ipc_mem_reg,
         .mem_dereg          = uct_cuda_ipc_mem_dereg,
+        .mem_query          = (uct_md_mem_query_func_t)ucs_empty_function_return_unsupported,
```

Comment:
is there an md_attr flag which indicates if mem_query function is supported?

Reply:
Not at the moment. If the transport is capable of memory detection (like cuda, rocm), then the transport can support mem_query where minimally it supports memory detection but it can also support system device detection. Eventually, we need transports like IB which are capable of system device detection alone (because we want to populate a list of system devices for topology query and population).

Reply:
let's add API flag for this function. We can later remote the mem_type_detect.